nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-2-6-processors/src/main/java/org/apache/nifi/processors/kafka/pubsub/PublishKafkaRecord_2_6.java

```diff
@@ -585,6 +586,10 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
                 failureStrategy.routeFlowFiles(session, flowFiles);
                 context.yield();
             }
+        } catch (final KafkaException e) {
+            getLogger().error("Failed to obtain Kafka publisher; will yield Processor", e);
```

Comment:
Recommend adjusting the message slightly since `Processor` will be included in the logger, and the operation also involves transferring files.
```suggestion
            getLogger().error("Failed to obtain Kafka Producer", e);
```

Reply:
Is the `KafkaException` only thrown when failing to obtain a Producer? This is a large block of code that seems like it could trigger other issues, but perhaps those are already handled?

Reply:
Good point.  What's a better way to handle that?

Reply:
One approach could be wrapping the `PublisherPool.obtainPublisher()` method in a separate method with the try-catch, and the return of that method takes the place of the current direct call to `obtainPublisher`. That separate method also needs to take into account the configured `Failure Strategy` and call that strategy method.

Reply:
The existing behavior doesn't consider the failure strategy unless a Kafka connection is initialized successfully.  Should we change that behavior?

Reply:
That's a good point, considering that the current exception forces a rollback. However, considering that the description of the `Fallback Strategy` seems to encompass all types of failures, honor the configuration seems like the best approach.

Reply:
I hoisted the call out into a helper method.